src/mpid/ch4/src/ch4_coll.h

```diff
@@ -13,49 +13,455 @@
+                              comm, errflag, bcast_container);
+      if (mpi_errno) MPIR_ERR_POP(mpi_errno);
+    }
+    else{
+      mpi_errno =
+          MPIDI_NM_mpi_bcast(buffer, count, datatype, root,
+                             comm, errflag, bcast_container);
+      if (mpi_errno) MPIR_ERR_POP(mpi_errno);
+    }
+#endif/*MPIDI_BUILD_CH4_SHM*/
+
+fn_exit:
+    return mpi_errno;
+fn_fail:
+    goto fn_exit;
+}
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_Bcast_intercomm
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int MPIDI_Bcast_intercomm(void *buffer, int count, MPI_Datatype datatype,
+                                                   int root, MPIR_Comm * comm, MPIR_Errflag_t * errflag,
+                                                   MPIDI_coll_algo_container_t * ch4_algo_parameters_container ATTRIBUTE((unused)))
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    mpi_errno = MPIR_Bcast_inter(buffer, count, datatype, root, comm, errflag);
+    if (mpi_errno) MPIR_ERR_POP(mpi_errno);
+
+fn_exit:
+    return mpi_errno;
+fn_fail:
+    goto fn_exit;
 }
 
 MPL_STATIC_INLINE_PREFIX int MPID_Allreduce(const void *sendbuf, void *recvbuf, int count,
                                              MPI_Datatype datatype, MPI_Op op, MPIR_Comm * comm,
                                              MPIR_Errflag_t * errflag)
 {
-    int ret;
+    int ret = MPI_SUCCESS;
+    MPIDI_coll_algo_container_t *ch4_algo_parameters_container = NULL;
 
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPID_ALLREDUCE);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPID_ALLREDUCE);
 
-    ret = MPIDI_NM_mpi_allreduce(sendbuf, recvbuf, count, datatype, op, comm, errflag);
+    ch4_algo_parameters_container =
+        MPIDI_CH4_Allreduce_select(sendbuf, recvbuf, count, datatype, op, comm, errflag);
+
+    ret = MPIDI_CH4_Allreduce_call(sendbuf, recvbuf, count, datatype, op, comm, errflag,
+                                   ch4_algo_parameters_container);
 
     MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPID_ALLREDUCE);
     return ret;
 }
 
+#undef FUNCNAME
+#define FUNCNAME MPIDI_Allreduce_composition_alpha
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int MPIDI_Allreduce_composition_alpha(const void *sendbuf, void *recvbuf, int count,
+                                                               MPI_Datatype datatype, MPI_Op op, MPIR_Comm * comm,
+                                                               MPIR_Errflag_t * errflag,
+                                                               MPIDI_coll_algo_container_t * ch4_algo_parameters_container)
+{
+    int mpi_errno = MPI_SUCCESS;
+    void * reduce_node_container = MPIDI_coll_get_next_container(ch4_algo_parameters_container);
+    void * allred_roots_container = MPIDI_coll_get_next_container(reduce_node_container);
+    void * bcast_node_container = MPIDI_coll_get_next_container(bcast_node_container);
```

Comment:
The compiler is complaining that `bcast_node_container` is used uninitialized here.
```
src/mpi/coll/allreduce.c: In function ‘MPIDI_Allreduce_composition_alpha’:
./src/mpid/ch4/src/ch4_coll.h:353:12: warning: ‘bcast_node_container’ is used uninitialized in this function [-Wuninitialized]
```